### ISSUE TITLE
Useless Condition Fix Produces Invalid Code

### ISSUE THREAD

**Issue by kiler129** [slevomat/coding-standard#617]
## Problem
With complicated from the static analysis perspective cases the `` sniff breaks the code on fix.

## Example
Before conversion the snippet below simply checks makes sure that two keys are either both empty or both filled:

```php
<?php
declare(strict_types=1);

function openHourDayIsValid($day): bool
{
    if ((
            empty($day['from']) && !empty($day['to'])
        ) || (
            !empty($day['from']) && empty($day['to'])
        )
    ) {
        return false;
    }

    return true;
}
```

After conversion the code is changed to:
```php
<?php
declare(strict_types=1);

function openHourDayIsValid($day): bool
{
    return (
            !empty($day['from']) || empty($day['to'])
        ) && (
            empty($day['from']) && empty($day['to'])
        )
    ;
}
```
They are not functionality equivalent ([proof](https://3v4l.org/O46rD)).

## Log
I ran `phpcbf` with `-vvv` and the result can be seen in [test.log](https://github.com/slevomat/coding-standard/files/2910928/test.log)

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/ac2ed16eb6490d1b54a8b0d195cb314723df5076
